### PR TITLE
Add state to filter only if its valid

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: ros/api/v1/hosts.py [hunk 1]
```diff
@@ -197,9 +197,10 @@ def build_system_filters():
             modified_states = []
             for state in states:
                 state = state.capitalize()
-                modified_states.append(state)
                 if state not in SYSTEM_STATES_EXCEPT_EMPTY:
                     abort(400, message='values are not matching')
+                else:
+                    modified_states.append(state)
```
Comment (kgaikwad) [inline ros/api/v1/hosts.py:203]: nitpick - `else` not necessary 

```suggestion
                if state not in SYSTEM_STATES_EXCEPT_EMPTY:
                    abort(400, message='values are not matching')
                modified_states.append(state)
```

File: ros/api/v1/hosts.py [hunk 2]
```diff
@@ -197,9 +197,10 @@ def build_system_filters():
             modified_states = []
             for state in states:
                 state = state.capitalize()
-                modified_states.append(state)
```
Comment (r14chandra) [inline ros/api/v1/hosts.py:200]: Previous and current changes both would work the same way. Is there any reason why we need this change?

Comment (saltgen) [reply ros/api/v1/hosts.py:200]: It is more of an optimization, looks like we can abort before an append operation

**Comment by saltgen** [comment]
@upadhyeammit Would you like to add a couple of more refactors on this PR?

File: ros/api/v1/hosts.py (hunk 2 above)
Comment (r14chandra) [reply ros/api/v1/hosts.py:200]: But append operation has constant time complexity.

Comment (upadhyeammit) [reply ros/api/v1/hosts.py:200]: I think we are doing unnecessary append to the list even if its invalid?

**Comment by upadhyeammit** [comment]
> @upadhyeammit Would you like to add a couple of more refactors on this PR?

no as of now

**Review by kgaikwad — APPROVED** [review]
Thank you @upadhyeammit! Merging this pull-request.

**Comment by kgaikwad** [comment]
Thank you @saltgen and @r14chandra for review.

**Pull request merged**